nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/ui/common/property-table/editors/nf-editor/nf-editor.component.ts

```diff
@@ -207,8 +218,18 @@ export class NfEditor implements OnDestroy {
 
     okClicked(): void {
         const valueControl: AbstractControl | null = this.nfEditorForm.get('value');
-        if (valueControl) {
-            this.ok.next(valueControl.value);
+        const emptyStringChecked: AbstractControl | null = this.nfEditorForm.get('setEmptyString');
+        if (valueControl && emptyStringChecked) {
+            const value = valueControl.value;
+            if (value === '') {
+                if (emptyStringChecked.value) {
+                    this.ok.next('');
+                } else {
+                    this.ok.next(null);
+                }
+            } else {
+                this.ok.next(value);
+            }
```

Comment:
Not sure where else to leave this comment, but we need to prevent `Enter` from submitting the okClicked action if the form is invalid (required value not entered)